gps-rtk-serial-no-network/gps_rtk_serial_no_network.go

```diff
@@ -290,43 +322,37 @@ func (g *rtkSerialNoNetwork) LinearVelocity(ctx context.Context, extra map[strin
 		return r3.Vector{}, lastError
 	}
 
-	return g.nmeamovementsensor.LinearVelocity(ctx, extra)
+	g.dataMu.RLock()
+	defer g.dataMu.RUnlock()
+	return r3.Vector{X: 0, Y: g.data.Speed, Z: 0}, g.err.Get()
 }
 
-// LinearAcceleration passthrough.
+// LinearAcceleration not supported.
 func (g *rtkSerialNoNetwork) LinearAcceleration(ctx context.Context, extra map[string]interface{}) (r3.Vector, error) {
-	lastError := g.err.Get()
-	if lastError != nil {
-		return r3.Vector{}, lastError
-	}
+	g.dataMu.RLock()
+	defer g.dataMu.RUnlock()
 	return r3.Vector{}, nil
 }
 
-// AngularVelocity passthrough.
+// AngularVelocity not supportd.
 func (g *rtkSerialNoNetwork) AngularVelocity(ctx context.Context, extra map[string]interface{}) (spatialmath.AngularVelocity, error) {
-	lastError := g.err.Get()
-	if lastError != nil {
-		return spatialmath.AngularVelocity{}, lastError
-	}
+	g.dataMu.RLock()
+	defer g.dataMu.RUnlock()
 
 	return spatialmath.AngularVelocity{}, nil
 }
 
-// CompassHeading passthrough.
+// CompassHeading not supported.
 func (g *rtkSerialNoNetwork) CompassHeading(ctx context.Context, extra map[string]interface{}) (float64, error) {
-	lastError := g.err.Get()
-	if lastError != nil {
-		return 0, lastError
-	}
-	return 0, nil
+	g.dataMu.RLock()
+	defer g.dataMu.RUnlock()
+	return 0, g.err.Get()
```

Comment:
Let's explicitly return unimplemented errors now in these two models.

Reply:
Running into an issue with this where when the movementsensor.go GetReadings function is called, its supposed to check if the error is unimplemented and if it is it just skips that reading, but since its an RPC error from the module it doesn't do that and I just get an error on the RC page and can't view any readings...is there a way around this?